### PR TITLE
IMAP Saved Search Results (RFC 5182)

### PR DESCRIPTION
**Pull request by ksmurchison**

### CODE REVIEW

File: lib/seqset.c [hunk 1]
```diff
@@ -139,6 +140,46 @@ EXPORTED void seqset_add(seqset_t *seq, unsigned num, int ismember)
     seq->prev = num;
 }
 
+/*
+ * Remove a number `num' from the sequence set `seq'.
+ * Currently assumes that it will be called in
+ * monotonically increasing order of `num's.
```
Comment (rsto) [inline lib/seqset.c:146]: This I find problematic. At least the code should assert that the input comes in the right order?

**Review by rsto — COMMENTED** [review]
Generally LGTM, but I'd prefer to sort out the seqset_remove function.

File: lib/seqset.c [hunk 2]
```diff
@@ -139,6 +140,46 @@ EXPORTED void seqset_add(seqset_t *seq, unsigned num, int ismember)
...
+        cur->high = num - 1;
+    }
+    else {
+        /* Split the range - insert a new one after current */
+        if (seq->len == seq->alloc) {
+            seq->alloc += SETGROWSIZE;
+            seq->set = xrealloc(seq->set, seq->alloc * RANGE_SIZE);
+        }
+
+        cur = &seq->set[seq->current];  // xrealloc may have changed ptr location
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply lib/seqset.c:174]: Actually, no.  We want to point to a specific element of the array, not the head

File: lib/seqset.c (hunk 1 above)
Comment (rsto) [reply lib/seqset.c:146]: Yeah! Sorry for the noise.

File: imap/imapd.c [hunk 3]
```diff
@@ -351,7 +351,7 @@ static struct capa_struct base_capabilities[] = {
     /* SASL-IR          RFC 4959 is announced in capa_response() */
     { "SAVEDATE",              2 }, /* RFC 8514 */
     { "SEARCH=FUZZY",          2 }, /* RFC 6203 */
-    /* SEARCHRES        RFC 5182 is not implemented */
+    { "SEARCHRES",             2 }, /* RFC 5182 */
     { "SORT",                  2 }, /* RFC 5256 */
```
[the remark this answers is not in the document]
Comment (ksmurchison) [reply imap/imapd.c:355]: Took care of 5182 in this PR.  WIll do 7377 separately

**Review by rsto — APPROVED** [review]
LGTM

**Pull request merged**